diff -ruN rtsp-linux/Makefile rtsp-linux-patched/Makefile
--- rtsp-linux/Makefile	2026-03-06 00:10:30
+++ rtsp-linux-patched/Makefile	2026-03-06 00:13:16
@@ -9,15 +9,40 @@
 
 # Normal Makefile
 
-KERNELDIR := /lib/modules/`uname -r`/build
-all: 
-	$(MAKE) -C $(KERNELDIR) M=`pwd` modules
+ifndef TARGET
+TARGET		:= $(shell uname -r)
+endif
 
+KERNEL_MODULES	:= /lib/modules/$(TARGET)
+
+ifneq ("","$(wildcard /usr/src/linux-headers-$(TARGET)/*)")
+KERNEL_BUILD	:= /usr/src/linux-headers-$(TARGET)
+else
+ifneq ("","$(wildcard /usr/src/kernels/$(TARGET)/*)")
+KERNEL_BUILD	:= /usr/src/kernels/$(TARGET)
+else
+KERNEL_BUILD	:= $(KERNEL_MODULES)/build
+endif
+endif
+
+DRIVER = nf_nat_rtsp
+MOD_SUBDIR = "net/ipv4/netfilter"
+MODDESTDIR=$(KERNEL_MODULES)/kernel/$(MOD_SUBDIR)
+
+all: build
+
+build:
+	$(MAKE) -C $(KERNEL_BUILD) M=`pwd` modules
+
 debug:
-	$(MAKE) -C $(KERNELDIR) EXTRA_CFLAGS=-DDEBUG M=`pwd` modules
+	$(MAKE) -C $(KERNEL_BUILD) EXTRA_CFLAGS=-DDEBUG M=`pwd` modules
 
+install: modules_install
+
 modules_install:
-	$(MAKE) -C $(KERNELDIR) M=`pwd` modules_install
+	mkdir -p $(MODDESTDIR)
+	cp nf_nat_rtsp.ko $(MODDESTDIR)/
+	cp nf_conntrack_rtsp.ko $(MODDESTDIR)/
 
 clean:
 	rm -rf *.o *.ko *.mod.c .*.cmd Module.symvers modules.order .tmp_versions
diff -ruN rtsp-linux/nf_conntrack_rtsp.c rtsp-linux-patched/nf_conntrack_rtsp.c
--- rtsp-linux/nf_conntrack_rtsp.c	2026-03-06 00:10:30
+++ rtsp-linux-patched/nf_conntrack_rtsp.c	2026-03-06 00:13:45
@@ -385,7 +385,7 @@
 		}
 
 		nf_nat_rtsp = rcu_dereference(nf_nat_rtsp_hook);
-		if (nf_nat_rtsp && ct->status & IPS_NAT_MASK)
+		if (nf_nat_rtsp && ct->status & IPS_NAT_MASK) {
 			/* pass the request off to the nat helper */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,7,0)
 			ret = nf_nat_rtsp(skb, ctinfo, protoff, hdrsoff, hdrslen,
@@ -394,7 +394,7 @@
 			ret = nf_nat_rtsp(skb, ctinfo, hdrsoff, hdrslen,
 					  &expinfo, rtp_exp, rtcp_exp);
 #endif
-		else {
+		} else {
 			if (rtsp_nf_ct_expect_related(rtp_exp) == 0) {
 				if (rtcp_exp && rtsp_nf_ct_expect_related(rtcp_exp) != 0) {
 					nf_ct_unexpect_related(rtp_exp);
@@ -471,6 +471,7 @@
 #endif
 
 	switch (CTINFO2DIR(ctinfo)) {
+	case IP_CT_DIR_MAX:
 	case IP_CT_DIR_ORIGINAL:
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3,7,0)
 		ret = help_out(skb, rb_ptr, datalen, ct, ctinfo, protoff);
